corehq/apps/hqadmin/management/commands/fix_checkpoints_from_date.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import print_function
+
+import argparse
+import difflib
+import pprint
+import re
+from datetime import datetime
+
+from django.core.management import BaseCommand, CommandError
+
+from corehq.apps.hqadmin.models import HistoricalPillowCheckpoint
+from pillowtop.models import str_to_kafka_seq
+from pillowtop.utils import get_all_pillow_instances
+
+
+def valid_date(s):
+    try:
+        return datetime.strptime(s, "%Y-%m-%d")
+    except ValueError:
+        msg = "Not a valid date: '{0}'.".format(s)
+        raise argparse.ArgumentTypeError(msg)
+
+
+def confirm(msg):
+    input = raw_input("{} Type ['y', 'yes'] to continue.\n".format(msg))
+    if input in ['y', 'yes']:
+        return True
+
+
+class Command(BaseCommand):
+    help = ("Update the pillow sequence IDs based on a passed in file")
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'date',
+            type=valid_date,
+            help="Take first checkpoint before date - format = YYYY-MM-DD",
+        )
+        parser.add_argument(
+            '--pillows',
+            nargs='+',
+            help="Restrict changes to only these pillows. (regex supported)"
+        )
+        parser.add_argument(
+            '--noinput',
+            action='store_true',
+            dest='noinput',
+            default=False,
+            help="Disable interactive mode",
+        )
+
+    def handle(self, **options):
```

Comment:
Should this have `date` as an argument since it's required?

Reply:
it can but it doesn't have to